blueprints/datamodel.tex

```diff
@@ -335,7 +335,117 @@ \subsection{Delivery Services}
 content and a set of Cache Servers and configuration options used to distribute
 that content.
 
-\subsubsection{Status}
+\subsubsection{Common Properties}
+Herein described are the properties common to all Delivery Service objects. The
+Routing Type of a Delivery Service encapsulates the methods by which clients
+may request content routing, and depending on its value the Delivery Service
+takes on a set of additional properties. Put simply, these are all different
+types of objects that are closely related.\\
+This section details all of the properties that are common to \emph{all} types of
+Delivery Services.
+
+\paragraph{Anonymous Blocking}
+A Delivery Service that has "Anonymous Blocking" tells Traffic Router to block
+requests from anonymized IP addresses. Whether or not and how well that can
+actually be done is dependent on the configuration of each Traffic Router itself,
+and if this Delivery Service is DNS-routed the only IP address Traffic Routers
+will be capable of checking for anonymization (e.g. known proxy/VPN/TOR exit node)
+then
+
+\paragraph{Deep Caching}
+
+\paragraph{Caching}
+\begin{itemize}
+	\item \code{CACHE}
+	\item \code{RAM\_ONLY}
+	\item \code{NO\_CACHE}
+\end{itemize}
+
+%TODO - STEERING doesn't have this?
+\paragraph{Cache Servers}
+A Delivery Service's assigned Cache Servers are expressed as a set of Cache Server
+IDs.
+
+\paragraph{CDN}
+
+%TODO: use-case for check path?
+
+\paragraph{Denied Access Redirect}
+
+\paragraph{DSCP}
+
+\paragraph{ECS}
+
+\paragraph{Geographic Limiting}
+This property describes limitations to the availability of this Delivery Service's
+content on the basis of the requesting client's geographic location. It is a set of
+strings, each of which is an \href{https://www.iso.org/obp/ui/#search/code/}{ISO 3166-1}
+alpha-2 country code, optionally with ISO 3166-2 subdivisional alphabetic code. This is
+a "white list" of countries/subdivisions wherein content is to be made available\footnote{
+This property is meant to inform Traffic Router; Cache Servers cannot be relied upon to
+approve or deny access on a geographic basis. Thus, if routing is bypassed, restricted
+content is totally accessible to requesting clients.}.\\
+Content is \emph{always} available to clients whose IP addresses are found within the
+Traffic Routers' Coverage Zone File(s). With that in mind, when this property is an empty
+set it means that no geographic regions are "whitelisted" and thus \emph{only} clients
+whose IP addresses are found within a Coverage Zone File will be granted access to content.
+When this property has a "Null" type, there is no geographic restriction placed on the
+Delivery Service's content access.
+
+\paragraph{Maximum Origin Connections}
+
+%TODO: use-case for header-rewrite text?
+
+\paragraph{Miss Location}
```

Comment:
This option should not be available to anyone without operator rights.  It's never correct to allow average users to specify what the default geo-miss location is or they could crush arbitrary cachegroups possibly.

Reply:
"Operator rights" will ideally be expressed as a set of user permissions that fall within the "operator" role. Is there a specific permission you have in mind, or do permissions not properly encapsulate the permission you are trying to express?

Reply:
Today the permission model applies to tables for the most part.  This field shouldn't be exposed to anyone without high permissions because of the potential for abuse.

Reply:
So you're saying that, in addition to the `deliveryservices-write` Permission someone  should also need the e.g. `deliveryservices-misslocation` permission to set the Miss Location of a Delivery Service to anything non-null?

Reply:
Yes